src/Rules.hs

```diff
@@ -44,7 +45,14 @@ buildLib stage pkg = do
             ways <- interpretInContext context getLibraryWays
             libs <- mapM (pkgLibraryFile . Context stage pkg) ways
             docs <- interpretInContext context $ buildHaddock flavour
-            need $ libs ++ [ pkgHaddockFile context | docs && stage == Stage1 ]
+
+            version <- interpretInContext context (getPkgData Version)
+            let topObj = buildPath context -/- "HS" ++
+                         pkgNameString pkg ++ "-" ++ version ++
+                         (waySuffix (way context) <.> "o")
```

Comment:
I think these objects are built for all libraries (I call them "ghci libraries", because as far as I understand they are required for `ghci`). Here is the build rule: 

https://github.com/snowleopard/hadrian/blob/master/src/Rules/Library.hs#L83-L89

Perhaps, the actual bug is here: 

https://github.com/snowleopard/hadrian/blob/master/src/Oracles/Dependencies.hs#L64-L67